src/app.ts

```diff
@@ -1,17 +1,44 @@
-import sslify from "express-sslify";
-import helmet from "helmet";
-import getFrontendApp from "./app";
-import { Application } from "probot";
-import { Express, NextFunction, Request, Response, Router } from "express";
+import express, { Express, NextFunction, Request, Response, Router } from "express";
+import path from "path";
+import getGithubClientMiddleware from "./middleware/github-client-middleware";
+import { App } from "@octokit/app";
+import { registerHandlebarsPartials } from "./util/handlebars/partials";
+import { registerHandlebarsHelpers } from "./util/handlebars/helpers";
 import crypto from "crypto";
-import Redis from "ioredis";
-import getRedisInfo from "../config/redis-info";
-import RateLimit from "express-rate-limit";
+import { Application } from "probot";
+import statsd, { elapsedTimeMetrics } from "./config/statsd";
+import IORedis from "ioredis";
+import getRedisInfo from "./config/redis-info";
+import rateLimit from "express-rate-limit";
 import RedisStore from "rate-limit-redis";
-import statsd, { elapsedTimeMetrics } from "../config/statsd";
-import { metricError } from "../config/metric-names";
+import { metricError } from "./config/metric-names";
+import sslify from "express-sslify";
+import helmet from "helmet";
+import { RootRouter } from "./routes/router";
+
+export const getFrontendApp = (octokitApp: App): Express => {
+	const app = express();
 
-function secureHeaders(router: Router, frontendApp: Express) {
+	// We run behind ngrok.io so we need to trust the proxy always
+	// TODO: look into the security of this.  Maybe should only be done for local dev?
+	app.set("trust proxy", true);
+	app.set("case sensitive routing", false);
+	app.set("view engine", "hbs");
+	const viewPath = path.resolve(process.cwd(), "views");
+	app.set("views", viewPath);
+	registerHandlebarsPartials(path.resolve(viewPath, "partials"));
+	registerHandlebarsHelpers();
+
+	// Add github client middleware which uses the octokit app
+	app.use(getGithubClientMiddleware(octokitApp));
```

Comment:
We have to have this here for now as it uses the octokit app to get github clients and such.  This will change in the future as we more away from probot/ocotkit.